### ISSUE TITLE
S2189 (Loops should not be infinite) raised on a variant condition

### ISSUE THREAD

**Issue by marcelubach** [SonarSource/SonarJS#1275]
**I want to report a bug.**

**SonarJS version:** 5.2.1 (build 7778)<br>
**SonarQube version:** 7.7<br>
**SonarLint version:** 1.9.0<br>

**Rule key:** javascript:S2189

**Reproducer** 
```
const sleep = time => {
  let count = 0;
  const waitTill = new Date(new Date().getTime() + time);
  while (waitTill > new Date()) { // False-positive here
    count += 1;
  }
};
```

**Expected behavior**
I think the _javascript:S2189_ issue should not be raised for the given code, The issue description recommends to "_Correct this loop's end condition as to not be invariant_" but really the condition changes since "New Date()" is different at each evaluation. 

For example, if we make this change explicit as shown in the code below, the issue is not raised any more:

const sleep = time => {
  let count = 0;
  const waitTill = new Date(new Date().getTime() + time);
  let now = new Date();
  while (waitTill > now) {
    count += 1;
    now = new Date();
  }
};
